Tools/peg_generator/pegen/python_generator.py

```diff
@@ -93,7 +93,13 @@ def visit_NegativeLookahead(self, node: NegativeLookahead) -> Tuple[None, str]:
 
     def visit_Opt(self, node: Opt) -> Tuple[str, str]:
         name, call = self.visit(node.node)
-        return "opt", f"{call},"  # Note trailing comma!
+        # Note trailing comma (if there isn't already one,
+        # which can occur when rules have both repeat0 and optional
+        # marker, e.g: [rule*])
+        if call.endswith(","):
```

Comment:
Hummmm, not a find on relying on the generated code to reason about the grammar structure (I know that we do it in other places but we have been removing those as it makes it more difficult to reason about). In this case I have to say that is pretty straightforward so it may be fine.